recipe/passwordless/passwordlessfunctions.go

```diff
@@ -30,6 +30,32 @@ import (
 var PasswordlessLoginEmailSentForTest bool = false
 var PasswordlessLoginSmsSentForTest bool = false
 
+func logAndReturnError(resp *http.Response, err error) error {
```

Comment:
How are these changes tested?

Reply:
```
func main() {
	configValue := supertokens.TypeInput{
		Supertokens: &supertokens.ConnectionInfo{
			ConnectionURI: "http://localhost:3567",
			APIKey:        "someKey",
		},
		AppInfo: supertokens.AppInfo{
			AppName:       "SuperTokens",
			WebsiteDomain: "localhost:3000",
			APIDomain:     "localhost:8000",
		},
		RecipeList: []supertokens.Recipe{
			passwordless.Init(plessmodels.TypeInput{
				FlowType: "USER_INPUT_CODE",
				ContactMethodEmail: plessmodels.ContactMethodEmailConfig{
					Enabled: true,
				},
			}),
		},
	}
	err := supertokens.Init(configValue)
	if err != nil {
		panic(err)
	}

	mux := http.NewServeMux()
	testServer := httptest.NewServer(supertokens.Middleware(mux))
	defer testServer.Close()

	unittesting.PasswordlessEmailLoginRequest("sattvik@gmail.com", testServer.URL)
}
```

Through debugging the above code. Similar for SMS as well.

Reply:
<img width="914" alt="image" src="https://user-images.githubusercontent.com/650429/171798561-220da486-e65e-4786-97aa-64ab5558fc37.png">

Debug logs received

Reply:
<img width="690" alt="image" src="https://user-images.githubusercontent.com/650429/171798681-f7757f75-3a8d-41c7-abb3-5668f8159174.png">

Returned error